src/hardware/node.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+namespace nest {
+namespace mc {
+namespace hw {
+
+// Information about the computational resources available on a compute node.
+// Currently a simple enumeration of the number of cpu cores and gpus, which
+// will become richer.
+struct node {
```

Comment:
I originally had `node_description` which I shortened when it was moved to the `hw` namespace. I will go with `node_info`.